### ISSUE TITLE
Timeline plugin incorrectly iterates over canvases

### ISSUE THREAD

**Issue by jimmyzzzzzzz** [katspaugh/wavesurfer.js#1222]
Is the change to the new for-in loops necessary? It would crash some existing scripts. For example I'm using my own array prototype attributes and it would enumerate through those and crash the script.  
I changed it back to the for i++ loop and it worked as before.

**Comment by mspae** [comment]
@jimmyzzzzzzz Can you post the lines in question here?

**Comment by jimmyzzzzzzz** [comment]
In wavesurfer.js/src/plugin/timeline.js
```
for (i in this.canvases) {
   const context = this.canvases[i].getContext('2d');
```
In my script I have something like
```
Array.prototype.test=function(){}
```

Then the loop will enumerate through this.canvases[0] and this.canvases['test'].

**Comment by mspae** [comment]
@jimmyzzzzzzz  This is actually not meant to be like this. It should use the `forEach` method because `this.canvases` is actually an array, not an object.

See https://github.com/katspaugh/wavesurfer.js/blob/master/src/plugin/timeline.js#L339

PRs welcome!

**Comment by thijstriemstra** [comment]
Fixed with #1224
